tls/extensions/s2n_extension_list.c

```diff
@@ -104,10 +104,6 @@ int s2n_extension_list_process(s2n_extension_list_id list_type, struct s2n_conne
                 conn, parsed_extension_list));
     }
 
-    /* If we did not process an extension, than that extension is not allowed on this message type. */
```

Comment:
Or maybe add the comment here about what we _would_ do if we were doing what the RFC suggests?